src/chainstate/stacks/index/storage.rs

```diff
@@ -920,62 +1053,69 @@ impl TrieFileStorage {
                     .open(&block_path)
                     .map_err(|e| {
                         if e.kind() == io::ErrorKind::NotFound {
-                            error!("File not found: {:?}", &block_path);
+                            trace!("File not found: {:?}", &block_path);
                             Error::NotFoundError
                         }
                         else {
                             Error::IOError(e)
                         }
                     })?;
 
+        // NOTE:
+        //   this doesn't really seem to improve performance.
```

Comment:
If that's the case, then please drop the type variant and just go with `fs::File`.

Reply:
I'm not that surprised, now that I think about it more.  The read I/O pattern would be random, so caching wouldn't really help.  Also, serialized nodes are small enough that they each fit inside an OS page, so it's questionable to me that any read-ahead scheme would speed up reading them.

Reply:
Yeah, I think it’s because a given file isn’t read repeatedly— file accesses will be much more random than that. I want to benchmark inserts after a large chain setup to confirm some of this.

Reply:
Addressed in da7f5c2 and 3b3ad513ead187283967e44aea3fe4f0bc6d5ca9

It just uses `fs::File` now. That second commit also gets rid of the option to open a block in readwrite mode -- writes can only occur if last_extended is set, and it points at the currently open block. (That commit also uses the type checker to enforce the last_extended_trie + last_extended header both being set at the same time but just turning it into a tuple).